dawn/src/dawn/Validator/UnstructuredDimensionChecker.cpp

```diff
@@ -169,6 +169,10 @@ void UnstructuredDimensionChecker::UnstructuredDimensionCheckerImpl::visit(
       dimensionsConsistent_ &= getUnstructuredDim(*curDimensions_).getDenseLocationType() ==
                                config_.currentChain_->back();
     }
+    if(hasOffset && !getUnstructuredDim(*curDimensions_).isDense()) {
+      dimensionsConsistent_ &=
+          getUnstructuredDim(*curDimensions_).getNeighborChain() == config_.currentChain_.value();
+    }
```

Comment:
The underlying assumption is that `hasOffset` implies `config_.currentChain_` being set (i.e. we are in a sparse iteration context). I wonder if such assumption is checked anywhere.